cpp/src/arrow/util/ree_util.h

```diff
@@ -138,6 +152,13 @@ int64_t FindPhysicalIndex(const ArraySpan& span, int64_t i, int64_t absolute_off
 /// end). This function uses binary-search, so it has a O(log N) cost.
 int64_t FindPhysicalLength(const ArraySpan& span);
 
+/// \brief Find the physical range of physical values referenced by the REE in
+/// the logical range from offset to offset + length
+///
+/// \return a pair of physical offset and physical length
+std::pair<int64_t, int64_t> FindPhysicalRange(const ArraySpan& span, int64_t offset,
```

Comment:
You need to ARROW_EXPORT this as well as the non-template `FindPhysicalIndex` and `FindPhysicalLength`, for Windows.

Reply:
Pushing a commit now.